docs/core/diagnostics/snippets/OTel-Prometheus-Grafana-Yaeger/csharp/Program.cs

```diff
@@ -0,0 +1,158 @@
+// <Snippet_HttpClientFactory>
+builder.Services.AddHttpClient();
+// </Snippet_HttpClientFactory>
+
+
+#if !AZURE_MONITOR
+// <Snippet_OTEL>
+var TracingOtlpEndpoint = builder.Configuration["OTLP_ENDPOINT_URL"];
+var otel = builder.Services.AddOpenTelemetry();
+
+// Configure OpenTelemetry Resources with the application name
+otel.ConfigureResource(resource => resource
+    .AddService(serviceName: builder.Environment.ApplicationName));
+
+// Add Metrics for ASP.NET Core and our custom metrics and export to Prometheus
+otel.WithMetrics(metrics => metrics
+    // Metrics provider from OpenTelemetry
+    .AddAspNetCoreInstrumentation()
+    .AddMeter(GreeterMeter.Name)
+    // Metrics provides by ASP.NET in .NET 8
+    .AddMeter("Microsoft.AspNetCore.Hosting")
+    .AddMeter("Microsoft.AspNetCore.Http.Connections")
+    .AddMeter("Microsoft.AspNetCore.Server.Kestrel")
+    .AddPrometheusExporter());
+
+// Add Tracing for ASP.NET Core and our custom ActivitySource and export to Jaeger
+otel.WithTracing(tracing =>
+{
+    tracing.AddAspNetCoreInstrumentation();
+    tracing.AddHttpClientInstrumentation();
+    tracing.AddSource(GreeterActivitySource.Name);
+    if (TracingOtlpEndpoint != null)
+    {
+        tracing.AddOtlpExporter(otlpOptions =>
+         {
+             otlpOptions.Endpoint = new Uri(TracingOtlpEndpoint);
+         });
+    }
+    else
+    {
+        tracing.AddConsoleExporter();
+    }
+});
+//</Snippet_OTEL>
+
+#else
+//<Snippet_AzureMonitor>
+var otel = builder.Services.AddOpenTelemetry();
+otel.UseAzureMonitor();
+otel.WithMetrics(metrics => metrics
+       .AddMeter(GreeterMeter.Name)
+       .AddMeter("Microsoft.AspNetCore.Hosting")
+       .AddMeter("Microsoft.AspNetCore.Http.Connections")
+       .AddMeter("Microsoft.AspNetCore.Server.Kestrel"));
+otel.WithTracing(tracing =>
+{
+    tracing.AddSource(GreeterActivitySource.Name);
+});
+//</Snippet_AzureMonitor>
+#endif
+
+var app = builder.Build();
+
+//<Snippet_MapGet>
+app.MapGet("/", SendGreeting);
+//</Snippet_MapGet>
+
+//<Snippet_MapNested>
+app.MapGet("/NestedGreeting", SendNestedGreeting);
+//</Snippet_MapNested>
+
+
+#if !AZURE_MONITOR
+//<Snippet_Prometheus>
+// Configure the Prometheus scraping endpoint
+app.UseOpenTelemetryPrometheusScrapingEndpoint();
+//</Snippet_Prometheus>
+#endif
+
+app.Run();
+
+//<Snippet_SendGreeting>
+async Task<String> SendGreeting(ILogger<Program> logger)
+{
+    // Create a new Activity scoped to the method
+    using var activity = GreeterActivitySource.StartActivity("GreeterActivity");
+
+    // Log a message
+    logger.LogInformation("Sending greeting");
+
+    // Increment the custom counter
+    CountGreetings.Add(1);
+
+    // Add a tag to the Activity
+    activity?.SetTag("greeting", "Hello World!");
+
+    return "Hello World!";
+}
+//</Snippet_SendGreeting>
+
+//<Snippet_SendNestedGreeting>
+async Task SendNestedGreeting(int nestlevel, ILogger<Program> logger, HttpContext context, IHttpClientFactory clientFactory)
```

Comment:
Is it a common practice to pass in logger instance to a business function? (I guess no)

Reply:
If the sample used MVC the typical pattern would be to assign ILogger to a field in the class, but in the minimal API style there is typically reduced structure and parameters are passed wherever needed. ILogger as parameter seems reasonable to me here.

Reply:
becuase this isn't a class, I don't have a constructor for DI to be injected into, so you do it at the method level.